platform/ios/src/MGLMapView.mm

```diff
@@ -1610,19 +1608,16 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
             [MGLMapboxEvents ensureMetricsOptoutExists];
         }
     }
-    else if ([keyPath isEqualToString:@"coordinate"] && [object conformsToProtocol:@protocol(MGLAnnotation)])
+    else if ([keyPath isEqualToString:@"coordinate"] && [object conformsToProtocol:@protocol(MGLAnnotation)] && ![object isKindOfClass:[MGLMultiPoint class]])
     {
         id <MGLAnnotation> annotation = object;
-        MGLAnnotationTag annotationTag = [self annotationTagForAnnotation:annotation];
-        if (annotationTag != MGLAnnotationTagNotFound)
+        MGLAnnotationTag annotationTag = (MGLAnnotationTag)(NSUInteger)context;
```

Comment:
Fixed in 1f23e4536d6719012886649cc6ad6fc97857437d.

Even if the context in that case happens to match the annotation’s tag – say, because the developer passed in `NULL` (which is equal to 0) for the first added annotation (which would have a tag of 0) – the result would be an unnecessary but harmless update. There could still be trouble around removing the observer, but this is such an edge case as to be programmer error.